component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -447,12 +450,11 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
         return isValidSignature;
     }
 
-    private String getValidAudience(String tenantDomain) throws OAuthClientAuthnException {
+    private List<String> getValidAudiences(String tenantDomain) throws OAuthClientAuthnException {
 
-        if (isNotEmpty(validAudience)) {
-            return validAudience;
-        }
+        List<String> validAudiences = new ArrayList<>();
         String audience = null;
```

Comment:
lets change this variable to reflect the actual name of the audience we are putting here as we are treating the parEndpoint also as a valid audience.

Reply:
Fixed in 5b618dc92cc5e65c0a0a7ee7a595a896cd31e0c2